docs/src/usage.md

```diff
@@ -31,4 +82,20 @@ julia> ca(FixedSizeArray{Float64}, iter)  # construct from an arbitrary iterator
 2-element FixedSizeArray{Float64, 1, Memory{Float64}}:
  7.0
  9.0
+
+julia> ca(FixedSizeVectorDefault, (3.14, -4.2, 2.68))  # construct from a tuple
+3-element FixedSizeArray{Float64, 1, Memory{Float64}}:
+  3.14
+ -4.2
+  2.68
 ```
+
+## `BoundsErrorLight` exception
+
+To facilitate the [escape analysis](https://en.wikipedia.org/wiki/Escape_analysis) of `FixedSizeArray`s, accessing an out-of-bound index of these arrays raises a [`BoundsErrorLight`](@ref) exception when possible.
+This exception type does not store the entire array for reporting the error message, thus enabling more performance optimizations compared to arrays which throw [`BoundsError`](https://docs.julialang.org/en/v1/base/base/#Core.BoundsError) exceptions.
+
+!!! warning "Compatibility of exception raised"
+
+    We do not guarantee to always throw `BoundsErrorLight` when accessing and out-of-bound index of a `FixedSizeArray`.
```

Comment:
While we throw `BoundsErrorLight` for the single-`Int`-index `getindex`, some `AbstractArray` fallbacks still throw `BoundsError`, as mentioned [here](https://github.com/JuliaArrays/FixedSizeArrays.jl/pull/116#issuecomment-2902027184). So I propose wording like this:

```suggestion
    We do not guarantee to throw either `BoundsError` or `BoundsErrorLight` when accessing an out-of-bound index of a `FixedSizeArray`.
```